openwisp_monitoring/device/apps.py

```diff
@@ -11,3 +15,57 @@ class DeviceMonitoringConfig(AppConfig):
 
     def ready(self):
         manage_short_retention_policy()
+        from openwisp_controller.connection.models import DeviceConnection
+        from openwisp_controller.connection.signals import is_working_changed
+        is_working_changed.connect(self.health_status_changed,
+                                   sender=DeviceConnection,
+                                   dispatch_uid='change_health_status')
+
+    @classmethod
+    def health_status_changed(cls, instance, is_working, **kwargs):
+        from ..check.models import Check
+        d = instance.device
+        dm = d.monitoring
+        initial_status = dm.status
+        status = 'ok' if is_working else 'problem'
```

Comment:
It is generally not recommended to write conditional blocks in one lines since it affects code coverage reports.

Reply:
Here, it stays unaffected. Still, I wasn't aware of it, thanks for sharing. Can you please share the source, I would like to know why so.

Reply:
this is ok